backend/src/Edna/Dashboard/Service.hs

```diff
@@ -64,10 +73,53 @@ setIsSuspiciousSubExperiment subExpId isSuspicious = do
 -- primary.
 deleteSubExperiment :: SubExperimentId -> Edna NoContent
 deleteSubExperiment subExpId = NoContent <$ do
+  -- Check existence (not the optimal way, but it likely doesn't matter)
+  void $ getMeasurements subExpId
```

Comment:
Not really understand what for you need it

Reply:
Without this check we fail with `DECantDeletePrimary` for non-existing sub-experiments.

![image](https://user-images.githubusercontent.com/2556409/114175149-581b2f80-9942-11eb-9815-49f6bf6ad331.png)

Reply:
Them looks like you again need transact here

Reply:
Ok, I'll add it.